### ISSUE TITLE
Locale selector for translator

### ISSUE THREAD

**Issue by lcharette** [userfrosting/UserFrosting#868]
The translator service can only work with one defined locale. We need a way to force a locale, for example when sending an email to a non-user. We can't simply use the logged-in user locale in that case.

**Comment by lcharette** [comment]
I started looking into this and I need your feedback on something. Here's my current plan. Basically, it's one locale per translator instance : 

```
$translator = new Translator('EN_us');
$translator->translate('HOME'); // Home

$translator_fr = new Translator('FR_fr');
$translator_fr->translate('HOME'); // Accueil
```

In this case, the Translator object can load and cache all keys for the selected locale. It will load all files for the defined locale, and would be able to return the locale metadata (#850; eg. `$translator->getLocaleAuthor()`) and so on. But it does change the whole behavior of the current translator, which the constructor accept [an array of keys](https://github.com/userfrosting/i18n/blob/c964f2dd4bb4ca66a1ea61ea94bf51cf0844a2d9/src/MessageTranslator.php#L40) generated by the [LocalePathBuilder](https://github.com/userfrosting/i18n/blob/c964f2dd4bb4ca66a1ea61ea94bf51cf0844a2d9/src/LocalePathBuilder.php#L35).

This solution would effectively merge the `Translator` and the `LocalePathBuilder` into the same class, but the Translator service would then only be able to translate one locale. So translating into a different locale would require a new instance (like the example above), or use a method to switch the overall locale (eg. `$translator->loadLocale('ES_es');`)

**Comment by amosfolz** [comment]
This would mean any missing or incomplete keys would no longer fall back to a "base" locale?

**Comment by lcharette** [comment]
No, as the fallback locale will be defined in the metadata.

**Comment by amosfolz** [comment]
I think it's a good plan then. Would the client-side code still work the same?

**Comment by lcharette** [comment]
Goal is for `->translate(...)` to stay the same

**Comment by lcharette** [comment]
This as been implemented with new version of i18n for UF 4.4
